cli/azd/pkg/project/importer.go

```diff
@@ -97,15 +100,31 @@ func (im *ImportManager) ServiceStable(ctx context.Context, projectConfig *Proje
 	return allServicesSlice, nil
 }
 
+// defaultOptions for infra settings. These values are applied across provisioning providers.
+const (
+	DefaultModule = "main"
+	DefaultPath   = "infra"
+)
+
+// ProjectInfrastructure parses the project configuration and returns the infrastructure configuration.
+// The configuration can be explicitly defined on azure.yaml using path and module, or in case these values
+// are not explicitly defined, the project importer uses default values to find the infrastructure.
 func (im *ImportManager) ProjectInfrastructure(ctx context.Context, projectConfig *ProjectConfig) (*Infra, error) {
+	// Use default project values for Infra when not specified in azure.yaml
+	if projectConfig.Infra.Module == "" {
```

Comment:
Long term maybe we can move to a place where the project config holds the defaults for these values and consumers never see a constructed one with "" for either `Module` or `Path`.

Reply:
that comes with a price because `ProjectConfig` is an exported structure.  It would require to be encapsulated (constructor + setters/getters) so it can ensure defaults. And usually, encapsulation does not feel idiomatic to go.

Maybe something idiomatic to go could be something like adding `project.withDefaults(p *ProjectConfig) *ProjectConfig` which delegates the  default-values assignation to a unique package-method.  

I'm happy to review more in the future about building/designing around the project package and how it is related to other packages.